// eslint-plugin-office-addins/lib/test/rules/no-office-initialize.test.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const rule_tester_1 = require("@typescript-eslint/rule-tester");
const no_office_initialize_1 = __importDefault(require("../../src/rules/no-office-initialize"));
const ruleTester = new rule_tester_1.RuleTester({
    parser: '@typescript-eslint/parser',
});
ruleTester.run('no-office-initialize', no_office_initialize_1.default, {
    valid: [
        {
            code: `Office.onReady();`
        },
        {
            code: `Office.onReady(function(info) {
          console.log(info);
        });`
        },
        {
            code: `Office.onReady()
          .then(function() {
            console.log("Testing Office.onReady followed by .then");
          });`
        },
        {
            code: `(async () => {
          await Office.onReady();
            console.log("Testing Office.onReady followed by await");
        })();`
        }
    ],
    invalid: [
        {
            code: `Office.initialize = function () {};`,
            errors: [{ messageId: "noOfficeInitialize" }]
        },
        {
            code: `Office.initialize = function () {
          console.log("Testing Office.initialize with normal function");
        };`,
            errors: [{ messageId: "noOfficeInitialize" }]
        },
        {
            code: `Office.initialize = function (reason) {
          $(document).ready(function () {
              console.log(reason);
          });
        };`,
            errors: [{ messageId: "noOfficeInitialize" }]
        },
        {
            code: `Office.initialize = () => {
          console.log("Testing Office.initialize with arrow function");
        };`,
            errors: [{ messageId: "noOfficeInitialize" }]
        }
    ]
});
